### PR TITLE
Storybook: Show code for Area component via StorySource in Examples and via Source in API 

### PR DESCRIPTION
**Pull request by nikolasrieble**
### The goal: 
We want to show the code in Storybook and enable users to copy paste both examples, as well as the result of using controls directly into their applications. 

### The tools: 
Storybook offers two ways to show code:

- [Storysource](https://storybook.js.org/addons/@storybook/addon-storysource) shows the complete code of the story. This works does not work well with using controls. 

- [Source](https://storybook.js.org/docs/react/api/doc-block-source#page-top) shows the code that is used for rendering the story. Works well with controls (i.e. does not show control setup, but only the value selected in the control). 

### The proposal:
Instead of using only one, lets use both, each where it fits best: 

- API  --> One story, in an MDX file, with an ArgsTable, showing the code via Source
- Examples  --> Multiple complex stories possibly with external dependencies, no MDX, no ArgsTable, showing the code via Storysource

### The result:

In this PR I implemented the proposal for the component `Area`:

We now have a single API story for the Area component, with an ArgsTable that shows all controls. The code can be shown in the MDX. 

Screenshot 2023-05-21 at 15 23 49
Screenshot 2023-05-21 at 15 24 10

In the Examples section, all Area usage stories went which are not easily usable with controls such as where functions are passed as arguments (for custom rendering of label/dot/etc.). Here, the story have no doc, and the code is visible in the story. 

Screenshot 2023-05-21 at 15 24 26

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by ckifer** [comment]
Just to clarify, this adds all of the stories from the `.stories` files to the bottom of the docs in a row?

Its better than not having them there at all (we can see the code now), but I think a separate file would be confusing if it was just some duplicated stories in a row. Maybe lets add them in this format to start and see how we feel about it for a little bit

**Comment by nikolasrieble** [comment]
> Just to clarify, this adds all of the stories from the `.stories` files to the bottom of the docs in a row?
> 
> Its better than not having them there at all (we can see the code now), but I think a separate file would be confusing if it was just some duplicated stories in a row. Maybe lets add them in this format to start and see how we feel about it for a little bit


Exactly. Stories need to be shown in a Canvas in an MDX document in order for the code to be shown. Do you have an alternative suggestion as to how this should be structured? 🤔

**Comment by nikolasrieble** [comment]
Lets not merge this yet. I want to look into the https://storybook.js.org/addons/@storybook/addon-storysource as an alternative before.

**Comment by nikolasrieble** [comment]
Updated this based on the discussions in slack. 
Looking for feedback: @ckifer @Yilun-Sun @proke03

File: storybook/stories/API/cartesian/Area.mdx [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
+<Controls of={ComponentStories.API} />
+
+
+
+
+
+
+
+
+
```
Comment (nikolasrieble) [inline storybook/stories/API/cartesian/Area.mdx:31]:
```suggestion








```

**Comment by nikolasrieble** [comment]
Next steps: If we agree on this pattern, I shall roll it out across the whole API section - component per component.

**Review by ckifer — APPROVED** [review]
Looks great! I think this pattern will be the best bet. Allows for great readability on the API side as well as great copy-paste resources / views of source for those trying to replicate an example. Thank you @nikolasrieble!

**Comment by nikolasrieble** [comment]
> Looks great! I think this pattern will be the best bet. Allows for great readability on the API side as well as great copy-paste resources / views of source for those trying to replicate an example. Thank you @nikolasrieble!

Great! I will then spend the next weekend 😅 (lets see whether that is enough) to migrate all existing stories to this pattern and update the README accordingly.

**Pull request merged**